airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/sponsored_display.py

```diff
@@ -12,11 +12,17 @@ class SponsoredDisplayCampaigns(SubProfilesStream):
     https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi#/Campaigns
     """
 
+    def __init__(self, *args, **kvargs):
+        state = kvargs.get("config", {}).get("stateFilter")
+        self.state_filter = ",".join(state) if state else ""
+        super().__init__(*args, **kvargs)
+
     primary_key = "campaignId"
     model = DisplayCampaign
 
     def path(self, **kvargs) -> str:
-        return "sd/campaigns"
+        params = f"?stateFilter={self.state_filter}" if self.state_filter else ""
```

Comment:
To append query parameters, override method [`request_params`](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L141). Method `path` is for the path part of the URL only.

Reply:
good catch! I'll implement that